core/src/main/java/io/grpc/MethodDescriptor.java

```diff
@@ -365,4 +367,130 @@ public static String extractFullServiceName(String fullMethodName) {
     }
     return fullMethodName.substring(0, index);
   }
+
+  /**
+   * Creates a new builder for a {@link MethodDescriptor}.
+   */
+  @CheckReturnValue
+  public static <ReqT, RespT> Builder<ReqT, RespT> newBuilder() {
+    return newBuilder(null, null);
+  }
+
+  /**
+   * Creates a new builder for a {@link MethodDescriptor}.
+   */
+  @CheckReturnValue
+  public static <ReqT, RespT> Builder<ReqT, RespT> newBuilder(
+      Marshaller<ReqT> requestMarshaller, Marshaller<RespT> responseMarshaller) {
+    return new Builder<ReqT, RespT>()
+        .setRequestMarshaller(requestMarshaller)
+        .setResponseMarshaller(responseMarshaller);
+  }
+
+  /**
+   * Turns this descriptor into a builder.
+   */
+  @CheckReturnValue
+  public Builder<ReqT, RespT> toBuilder() {
+    return toBuilder(requestMarshaller, responseMarshaller);
+  }
+
+  /**
+   * Turns this descriptor into a builder, replacing the request and response marshallers.
+   */
+  @CheckReturnValue
+  public <NewReqT, NewRespT> Builder<NewReqT, NewRespT> toBuilder(
+      Marshaller<NewReqT> requestMarshaller, Marshaller<NewRespT> responseMarshaller) {
+    return MethodDescriptor.<NewReqT, NewRespT>newBuilder()
+        .setRequestMarshaller(requestMarshaller)
+        .setResponseMarshaller(responseMarshaller)
+        .setType(type)
+        .setFullMethodName(fullMethodName)
+        .setIdempotent(idempotent)
+        .setSafe(safe);
+  }
+
+  /**
+   * A builder for a {@link MethodDescriptor}.
+   */
+  public static final class Builder<ReqT, RespT> {
```

Comment:
`MethodDescriptor.create()` isn't going away, so I don't see what the rush is to avoid using it in generated code. And I do have concern with committing a new API and releasing it as stable a couple days later.

> Also, I would probably have to back out all the changes from the non-code packages.

What do you mean? non-code?